cosmwasm/main.go

```diff
@@ -0,0 +1,280 @@
+package main
+
+import (
+	"embed"
+	"encoding/gob"
+	"fmt"
+	"io"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	hplugin "github.com/hashicorp/go-plugin"
+
+	"github.com/ignite/cli/ignite/services/plugin"
+)
+
+//go:embed wasm-wiring/*
+var templates embed.FS // Embedded template files
+
+func init() {
+	gob.Register(plugin.Manifest{})
+	gob.Register(plugin.ExecutedCommand{})
+	gob.Register(plugin.ExecutedHook{})
+}
+
+type p struct {
+	chainName string
+}
+
+// NewPlugin creates a new plugin instance with the given chainName and chainNamed values.
+func NewPlugin(chainName string) *p {
+	return &p{
+		chainName: chainName,
+	}
+}
+
+// ChainName returns the chainName of the plugin.
+func (p *p) ChainName() string {
+	return p.chainName
+}
+
```

Comment:
The `ChainName` method can be removed because is not being used:

```suggestion
```

Reply:
Resolved: https://github.com/yerasyla/plugins/commit/27f2900b6c853a13a1269fa3d74fe471399bc399